core/state/database.go

```diff
@@ -54,6 +54,9 @@ type Database interface {
 	// ContractCodeSize retrieves a particular contracts code's size.
 	ContractCodeSize(addrHash, codeHash common.Hash) (int, error)
 
+	// DiskDB returns the underlying key-value disk database.
+	DiskDB() ethdb.KeyValueStore
```

Comment:
The only usage of this (aside from tests) is `codeWriter := s.db.DiskDB().NewBatch()` in `statedb.go`, when we want to flush code directly to disk without having it cached in the triedb database. 
So if we want to make the interface even smaller, we could change this to e.g. `NewDiskBatch() ethdb.Batch`

Reply:
What about `WriteCode` interface?

we can call it like `s.db.WriteCode(common.BytesToHash(obj.CodeHash()), obj.code)`. Although performance wise, it will have some tiny writes but since contract creation is not that frequent, I guess no big performance hit.